djangomako/backends.py

```diff
@@ -164,7 +165,37 @@ def render(self, context=None, request=None):
             context['csrf_input'] = csrf_input_lazy(request)
             context['csrf_token'] = csrf_token_lazy(request)
 
-        return self.template.render(**context)
+        try:
+            return self.template.render(**context)
+        except Exception as e:
+            tb = RichTraceback()
+
+            lines = tb.source
+            if not lines:
+                # There's no templates lines then raise
+                raise e
+
+            lines = lines.split('\n')
+            line = tb.lineno
+            top = max(0, line - 4)
+            bottom = min(len(lines), line + 5)
+            source_lines = [(i+1, lines[i] + '\n') for i in range(top, bottom)]
+
+            e.template_debug = {
+                'name': tb.traceback[len(tb.traceback) - 1][0],
+                'message': '{}: {}'.format(tb.errorname, tb.message),
+                'source_lines': source_lines,
+                'line': line,
+                'during': source_lines[line - top - 1][1],
```

Comment:
let's set a variable to equal `source_lines[error_line - top - 1][1]` to make the code more readable.

Reply:
Not sure how much it will make it more readable, it's a dict then it's explicitly named